src/Microsoft.SqlTools.ServiceLayer/EditData/EditDataService.cs

```diff
@@ -0,0 +1,328 @@
+                await queryExecutionService.InterServiceExecuteQuery(executeParams, requestContext,
+                    queryCreateSuccessCallback, queryCreateFailureCallback,
+                    queryCompleteSuccessCallback, queryCompleteFailureCallback);
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e.Message);
+            }
+        }
+
+        internal async Task HandleRevertRowRequest(EditRevertRowParams revertParams,
+            RequestContext<EditRevertRowResult> requestContext)
+        {
+            try
+            {
+                Session session = GetActiveSessionOrThrow(revertParams.OwnerUri);
+                session.RevertRow(revertParams.RowId);
+                await requestContext.SendResult(new EditRevertRowResult());
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e.Message);
+            }
+        }
+
+        internal async Task HandleUpdateCellRequest(EditUpdateCellParams updateParams,
+            RequestContext<EditUpdateCellResult> requestContext)
+        {
+            try
+            {
+                Session session = GetActiveSessionOrThrow(updateParams.OwnerUri);
+                var result = session.UpdateCell(updateParams.RowId, updateParams.ColumnId, updateParams.NewValue);
+                await requestContext.SendResult(result);
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e.Message);
+            }
+        }
+
+        #endregion
+
+        #region Private Helpers
+
+        /// <summary>
+        /// Returns the session with the given owner URI or throws if it can't be found
+        /// </summary>
+        /// <exception cref="Exception">If the edit session doesn't exist</exception>
+        /// <param name="ownerUri">Owner URI for the edit session</param>
+        /// <returns>The edit session that corresponds to the owner URI</returns>
+        private Session GetActiveSessionOrThrow(string ownerUri)
+        {
+            // Sanity check the owner URI is provided
+            Validate.IsNotNullOrWhitespaceString(nameof(ownerUri), ownerUri);
+
+            // Attempt to get the session, throw if unable
+            Session session;
+            if (!ActiveSessions.TryGetValue(ownerUri, out session))
+            {
+                throw new Exception(SR.EditDataSessionNotFound);
+            }
+
+            return session;
+        }
+
+        #endregion
+
+        #region IDisposable Implementation
+
+        private bool disposed;
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                // TODO: Dispose objects that need disposing
```

Comment:
... I guess nothing. I thought I needed to implement a IDisposable for the Session, but there isn't any resources that need disposing in there.